### DISCUSSION TITLE
개인 프로필 API 설계에 대한 고민

### DISCUSSION THREAD

**Discussion by openmpy** [two-clock/backend#33] [General]
@jjangsky 님.
GitConnect 서비스에서 개인 프로필 정보를 어떻게 가져올지에 대해 두 가지 접근 방식을 고려하고 있습니다.

1. GitHub API를 직접 호출하여 프로필 정보 가져오기
- 장점: 추가적인 작업 없이, 즉시 GitHub에서 최신 정보를 가져올 수 있습니다. 구현이 간단합니다.
- 단점: GitHub 서비스에 대한 의존성이 높아집니다. 이용자가 많아지면 GitHub API 호출 횟수가 증가해 API 제한에 걸릴 위험이 있습니다. 또한, GitHub API의 변경사항이나 장애에 직접적인 영향을 받을 수 있습니다.
2. DB에서 정보 조회 후, 필요 시 GitHub API 호출하기
- 장점: GitHub 서비스에 대한 의존성이 줄어들며, API 호출을 최소화할 수 있습니다. 서비스 확장 시 유연성이 높아지고, 캐싱된 데이터를 사용해 성능을 최적화할 수 있습니다.
- 단점: ERD 수정 및 추가적인 데이터베이스 작업이 필요합니다. 데이터의 최신성을 유지하기 위한 별도의 동기화 로직이 필요할 수 있습니다.

어떤 방식이 더 나을지, 혹은 다른 좋은 아이디어가 있는지 의견 부탁드립니다.

**Comment by jjangsky** [comment]
@openmpy 님 저의 소견을 말씀드리자면
개인적으로 2번 방식을 더 선호하는 편입니다.

1번 방식처럼 로그인 할 때마다 혹은 특정 작업 마다 Github API를 사용하여 정보를 계속 갱신하면 좋겠다만,
말씀하신 것처럼 과도한 API 요청으로 Github로부터 API 차단 가능성을 염두해야 할 것 같습니다.

저희 서비스가 Github에 의존하는 서비스는 맞지만 프로필에 중점적일 필요는 없으니 최초 로그인 혹은 회원가입 시점에서만
정보를 받아오고 이후 프로필 서비스 수정 기능을 통해 자신만의 프로필을 만들 수 있는 기능도 괜찮을 것 같습니다.
아니면 Github 프로필 갱신 기능 이런거 추가해서 사용자가 원할 때 동기화 할 수 있도록 기능을 넣어도 괜찮을 것 같습니다.

**Reply by openmpy** [reply]
@jjangsky 님 질문에 대한 상세한 답변 감사드립니다.
덕분에 방향성을 잡고 작업을 진행할 수 있을거 같습니다.
해당 토론은 닫도록 하겠습니다 ~
